Userland/Applications/HexEditor/HexEditor.cpp

```diff
@@ -409,6 +430,9 @@ void HexEditor::hex_mode_keydown_event(GUI::KeyEvent& event)
     if ((event.key() >= KeyCode::Key_0 && event.key() <= KeyCode::Key_9) || (event.key() >= KeyCode::Key_A && event.key() <= KeyCode::Key_F)) {
         if (m_buffer.is_empty())
             return;
+        if (m_position == static_cast<int>(m_buffer.size()))
```

Comment:
This should probably be size_t, as with all data structure sizes.

Reply:
Not sure if I understand correctly. You mean `m_position` should be size_t?
Cause that's a heckin lot of changes.

There's multiple places in HexEditor that do different signedness comparisons and
not all of them would make sense to rewrite as unsigned. It would require a lot of
explicit checks for negative values.

Reply:
If m_position is not a size_t, yes, that's a lot of changes, but remember that size_t is unsigned, and the buffer size should be size_t. Casting to int leads to a lot of nasty problems because int has only half the positive range of size_t. I think we need a third opinion on this. (Also, a negative m_position makes no sense, so...)

Reply:
Yes that makes sense. I changed some variables to be size_t. However the rendering code still has some issues that would make rendering very big files not possible, like being dependent on absolute row numbers. There are a number of other things I would change before tackling that, though. I think the most important feature would be a proper editing structure to allow deleting and inserting of bytes, as well as undo and redo.

Reply:
That's great! Such changes would be its own yak, so I don't expect it here. I'll have another look later and we're probably good then.